Numpy math methods (#4513)
Codecademy/docs: content/numpy/concepts/math-methods/math-methods.md

> 

Reviewer: ```suggestion
---
Title: 'Math Methods'
Description: 'Performs mathematical operations on arrays'
Subjects:
  - 'Computer Science'
Tags:
  - 'NumPy'
  - 'Math'
  - 'Methods'
  - 'Arrays'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---
```

> # Math Methods

Reviewer: ```suggestion
Remove the heading with # completely, since it is not needed
```

> # Math Methods

Reviewer: Please remove any headings with #

>   - Numerical Python abreviated as numPy, used for numerical computing.
>   - It provides a wide range of mathematical functions and operations on scalar values, arrays and matrices.
>   - Acts as a fundamental tool for scientific computing, data analysis, and machine learning.
>   - NumPy's math methods facilitate complex mathematical computations efficiently, making it indispensable in various domains including scientific research, engineering, finance, and data analysis.
>   - Understanding these methods empowers practitioners to leverage the full potential of NumPy for their numerical computing tasks.

Reviewer: ```suggestion
  Avoid writing in bullet points, rather convert the points into a paragraph format, an example could be:
  
 **NumPy**, short for Numerical Python, is a powerhouse for numerical computing, offering many mathematical functions and operations on scalars, arrays, and matrices. It plays a pivotal role in scientific computing, data analysis, and machine learning, serving as an essential tool in various fields like scientific research, engineering, finance, and data analysis. NumPy's mathematical methods streamline complex computations, making it an indispensable asset. Mastering these methods empowers users to harness NumPy's full potential for their numerical computing endeavors.
```

>   - Understanding these methods empowers practitioners to leverage the full potential of NumPy for their numerical computing tasks.

Reviewer: Avoid writing in bullet points, rather convert the points into a paragraph format, an example could be:
  
 **NumPy**, short for Numerical Python, is a powerhouse for numerical computing, offering many mathematical functions and operations on scalars, arrays, and matrices. It plays a pivotal role in scientific computing, data analysis, and machine learning, serving as an essential tool in various fields like scientific research, engineering, finance, and data analysis. NumPy's mathematical methods streamline complex computations, making it an indispensable asset. Mastering these methods empowers users to harness NumPy's full potential for their numerical computing endeavors.

You can modify it according to you, but design it in a paragraph format.

> ## Syntax

Reviewer: Please give a SINGLE LINE space before and after every topic.

>  numpy. <math_method>

Reviewer: ```suggestion
 Please use the following for representing syntax:
 
 ```pseudo 
 numpy.<math_method>
```
```

## Example

> ![image](https://github.com/gitsoft145/docs/assets/78934620/7f0cf44a-f91d-4ef2-95e5-7cb065387980)

Reviewer: ```suggestion
Instead of using an image, add the code block here, as follows:

```pseudo
import numpy as np

//add two scalar numbers using numpy.add() method
result = np.add(5, 3)

print("Result of addition:", result)
```

```

> 

Reviewer: ```
Leave only a SINGLE LINE space after each topic.
```

> 

Reviewer: Make sure you only leave a SINGLE LINE space after every topic.

## Output

> ![image](https://github.com/gitsoft145/docs/assets/78934620/cb0b3d5b-7c91-45a5-b301-3d56cce64bb0)

Reviewer: ```suggestion
For the output as well, use the code block under the ``` psuedo block
```

> ![image](https://github.com/gitsoft145/docs/assets/78934620/cb0b3d5b-7c91-45a5-b301-3d56cce64bb0)

Reviewer: For the output as well, please use the ```pseudo block
please refer to the [Style Guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md) for better understanding.

> 

Reviewer: Leave only a SINGLE LINE space after each topic.

> 

Reviewer: Make sure you only leave a SINGLE LINE space after every topic.

## Basic Categories of Math Methods

  ### 1. Basic Arithmetic Operations
  

>       - Addition (numpy.add): Element-wise addition of arrays and scalar values.
>       - Subtraction (numpy.subtract): Element-wise subtraction of arrays.
>       - Multiplication (numpy.multiply): Element-wise multiplication of arrays.
>       - Division (numpy.divide): Element-wise division of arrays.
>       - Power (numpy.power): Element-wise exponentiation of arrays.

Reviewer: ```suggestion
Only two spaces are needed before every point for the bullets.
For all the points listed above, wrap the syntax in the backticks and use the word "Performs" before every sentence. 
Example: 
  - Addition `numpy.add()`: Performs element-wise addition of arrays and scalar values.
```

>       - Power (numpy.power): Element-wise exponentiation of arrays.

Reviewer: Only two spaces are needed before every point for the bullets.
For all the points listed above, wrap the syntax in the backticks and use the word "Performs" before every sentence. 
Example: 
  - Addition `numpy.add()`: Performs element-wise addition of arrays and scalar values.

      
  ### 2. Trigonometric Functions
  

>       - Sine (numpy.sin): Computes the sine of each element in the array.
>       - Cosine (numpy.cos): Computes the cosine of each element in the array.
>       - Tangent (numpy.tan): Computes the tangent of each element in the array.
>       - Inverse Sine (numpy.arcsin): Computes the inverse sine of each element in the array.
>       - Inverse Cosine (numpy.arccos): Computes the inverse cosine of each element in the array.
>       - Inverse Tangent (numpy.arctan): Computes the inverse tangent of each element in the array.

Reviewer: ```suggestion
Only two spaces are needed before every point for the bullets.
For all the points listed above, wrap the syntax in the backticks. Example:
   - Sine `numpy.sin()`: Computes the sine of each element in the array.
```

>       - Inverse Tangent (numpy.arctan): Computes the inverse tangent of each element in the array.

Reviewer: Only two spaces are needed before every point for the bullets.
For all the points listed above, wrap the syntax in the backticks. Example:
   - Sine `numpy.sin()`: Computes the sine of each element in the array.

      
  ### 3. Exponential and Logarithmic Functions
  

>       - Exponential (numpy.exp): Computes the exponential of each element in the array.
>       - Natural Logarithm (numpy.log): Computes the natural logarithm of each element in the array.
>       - Base-10 Logarithm (numpy.log10): Computes the base-10 logarithm of each element in the array.
>       - Logarithm with Base b (numpy.log): Computes the logarithm with specified base of each element in the array.

Reviewer: ```suggestion
  Only two spaces are needed before every point for the bullets.
  For all the points listed above, wrap the syntax in the backticks. 
```

>       - Logarithm with Base b (numpy.log): Computes the logarithm with specified base of each element in the array.

Reviewer: ```suggestion
  - Logarithm with Base b (numpy.log): Computes the logarithm with the specified base of each element in the array.
```

>       - Logarithm with Base b (numpy.log): Computes the logarithm with specified base of each element in the array.

Reviewer: Only two spaces are needed before every point for the bullets.
For all the points listed above, wrap the syntax in the backticks.

      
  ### 4. Miscellaneous Functions
  

>       - Absolute Value (numpy.abs): Computes the absolute value of each element in the array.
>       - Square Root (numpy.sqrt): Computes the square root of each element in the array.
>       - Ceiling (numpy.ceil): Rounds each element of the array to the nearest integer greater than or equal to that element.
>       - Floor (numpy.floor): Rounds each element of the array to the nearest integer less than or equal to that element.
>       - Rounding (numpy.round): Rounds each element of the array to the nearest integer.

Reviewer: ```suggestion
Only two spaces are needed before every point for the bullets.
For all the points listed above, wrap the syntax in the backticks.
```

>       - Rounding (numpy.round): Rounds each element of the array to the nearest integer.

Reviewer: Only two spaces are needed before every point for the bullets.
For all the points listed above, wrap the syntax in the backticks.

      
  ### 5. Statistical Functions
  

>       - Mean (numpy.mean): Computes the arithmetic mean along the specified axis.
>       - Median (numpy.median): Computes the median along the specified axis.
>       - Standard Deviation (numpy.std): Computes the standard deviation along the specified axis.
>       - Variance (numpy.var): Computes the variance along the specified axis.

Reviewer: ```suggestion
Only two spaces are needed before every point for the bullets.
For all the points listed above, wrap the syntax in the backticks.
```

>       - Variance (numpy.var): Computes the variance along the specified axis.

Reviewer: Only two spaces are needed before every point for the bullets.
For all the points listed above, wrap the syntax in the backticks.

  
  ### 6. Linear Algebra
  

>       - Matrix Multiplication (numpy.matmul): Performs matrix multiplication between two arrays.
>       - Dot Product (numpy.dot): Computes the dot product of two arrays.
>       - Transpose (numpy.transpose): Computes the transpose of an array.

Reviewer: ```suggestion
Only two spaces are needed before every point for the bullets.
For all the points listed above, wrap the syntax in the backticks.
```

>       - Transpose (numpy.transpose): Computes the transpose of an array.

Reviewer: Only two spaces are needed before every point for the bullets.
For all the points listed above, wrap the syntax in the backticks.

Author: I have made all changes

> 
> 
> 
> 

Reviewer: ```suggestion
The four extra lines need to be removed from here.
```

> 

Reviewer: Please remove these extra 4 lines, they are not needed.

---
Title: 'Math Methods'
Description: 'Performs mathematical operations on arrays'
Subjects:
  - 'Computer Science'
Tags:
  - 'NumPy'
  - 'Math'
  - 'Methods'
  - 'Arrays'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

> # Numpy Math Methods
> ** Description: **

Reviewer: ```suggestion
```

Author: I have made the changes

> NumPy, short for Numerical Python, is a powerhouse for numerical computing, offering many mathematical functions and operations on scalars, arrays, and matrices. It plays a pivotal role in scientific computing, data analysis, and machine learning, serving as an essential tool in various fields like scientific research, engineering, finance, and data analysis. NumPy's mathematical methods streamline complex computations, making it an indispensable asset. Mastering these methods empowers users to harness NumPy's full potential for their numerical computing endeavors.

Reviewer: ```suggestion
NumPy, or Numerical Python, is a powerhouse for numerical computing, offering an extensive range of mathematical methods known as **Math Methods**. These methods cover arithmetic operations, trigonometric functions, linear algebra, statistics, and more. They are essential in scientific computing, data analysis, and machine learning, making NumPy indispensable across scientific research, engineering, finance, and data analysis. Mastering NumPy's Math Methods empowers users to streamline complex computations effectively
```

## Syntax

The generic syntax for the NumPy math methods is as follows:
 
 ```pseudo 
 numpy.<math_method>

> 

Reviewer: ```suggestion
```
```

---
Title: 'Math Methods'
Description: 'Performs mathematical operations on arrays'
Subjects:
  - 'Computer Science'
Tags:
  - 'NumPy'
  - 'Math'
  - 'Methods'
  - 'Arrays'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

> 
> # Math Methods

Reviewer: ```suggestion

```

>   - Numerical Python abreviated as numPy, used for numerical computing.
>   - It provides a wide range of mathematical functions and operations on scalar values, arrays and matrices.
>   - Acts as a fundamental tool for scientific computing, data analysis, and machine learning.
>   - NumPy's math methods facilitate complex mathematical computations efficiently, making it indispensable in various domains including scientific research, engineering, finance, and data analysis.
>   - Understanding these methods empowers practitioners to leverage the full potential of NumPy for their numerical computing tasks.

Reviewer: ```suggestion
The **`Math methods`** available in `NumPy` (short for Numerical Python) are incredibly powerful when it comes to numerical computing. They offer a wide range of mathematical functions and operations that can be applied to scalars, arrays, and matrices. These methods are essential in many fields, including scientific research, engineering, finance, and data analysis. Using NumPy's mathematical methods can streamline complex computations, making them an indispensable asset. By mastering these methods, you can take full advantage of NumPy's potential for all your numerical computing needs.
```

## Syntax

The generic syntax for the NumPy math methods is as follows:
 

>  ```pseudo 
>  numpy.<math_method>

Reviewer: ```suggestion
 ```pseudo 
 numpy.<math_method>
 ```
```

> ## Example

Reviewer: ```suggestion
## Example

```

```py
import numpy as np

//add two scalar numbers using numpy.add() method
result = np.add(5, 3)

print("Result of addition:", result)

## Output

> ![image](https://github.com/gitsoft145/docs/assets/78934620/cb0b3d5b-7c91-45a5-b301-3d56cce64bb0)
> 

Reviewer: ```suggestion

```shell
Result of addition: 8
```
```

## Basic Categories of Math Methods

  ### 1. Basic Arithmetic Operations
  

>       - Addition (numpy.add): Element-wise addition of arrays and scalar values.
>       - Subtraction (numpy.subtract): Element-wise subtraction of arrays.
>       - Multiplication (numpy.multiply): Element-wise multiplication of arrays.
>       - Division (numpy.divide): Element-wise division of arrays.
>       - Power (numpy.power): Element-wise exponentiation of arrays.

Reviewer: ```suggestion
  - Addition `numpy.add`: Performs element-wise addition of arrays and scalar values.
  - Subtraction `numpy.subtract`: Performs element-wise subtraction of arrays.
  - Multiplication `numpy.multiply`: Performs element-wise multiplication of arrays.
  - Division `numpy.divide`: Performs element-wise division of arrays.
  - Power `numpy.power`: Performs element-wise exponentiation of arrays.
```

      
  ### 2. Trigonometric Functions
  

>       - Sine (numpy.sin): Computes the sine of each element in the array.
>       - Cosine (numpy.cos): Computes the cosine of each element in the array.
>       - Tangent (numpy.tan): Computes the tangent of each element in the array.
>       - Inverse Sine (numpy.arcsin): Computes the inverse sine of each element in the array.
>       - Inverse Cosine (numpy.arccos): Computes the inverse cosine of each element in the array.
>       - Inverse Tangent (numpy.arctan): Computes the inverse tangent of each element in the array.

Reviewer: ```suggestion
  - Sine `numpy.sin`: Computes the sine of each element in the array.
  - Cosine `numpy.cos`: Computes the cosine of each element in the array.
  - Tangent `numpy.tan`: Computes the tangent of each element in the array.
  - Inverse Sine `numpy.arcsin`: Computes the inverse sine of each element in the array.
  - Inverse Cosine `numpy.arccos`: Computes the inverse cosine of each element in the array.
  - Inverse Tangent `numpy.arctan`: Computes the inverse tangent of each element in the array.
```

      
  ### 3. Exponential and Logarithmic Functions
  

>       - Exponential (numpy.exp): Computes the exponential of each element in the array.
>       - Natural Logarithm (numpy.log): Computes the natural logarithm of each element in the array.
>       - Base-10 Logarithm (numpy.log10): Computes the base-10 logarithm of each element in the array.
>       - Logarithm with Base b (numpy.log): Computes the logarithm with specified base of each element in the array.

Reviewer: ```suggestion
  - Exponential `numpy.exp`: Computes the exponential of each element in the array.
  - Natural Logarithm `numpy.log`: Computes the natural logarithm of each element in the array.
  - Base-10 Logarithm `numpy.log10`: Computes the base-10 logarithm of each element in the array.
  - Logarithm with Base b `numpy.log`: Computes the logarithm with the specified base of each element in the array.
```

      
  ### 4. Miscellaneous Functions
  

>       - Absolute Value (numpy.abs): Computes the absolute value of each element in the array.
>       - Square Root (numpy.sqrt): Computes the square root of each element in the array.
>       - Ceiling (numpy.ceil): Rounds each element of the array to the nearest integer greater than or equal to that element.
>       - Floor (numpy.floor): Rounds each element of the array to the nearest integer less than or equal to that element.
>       - Rounding (numpy.round): Rounds each element of the array to the nearest integer.

Reviewer: ```suggestion
  - Absolute Value `numpy.abs`: Computes the absolute value of each element in the array.
  - Square Root `numpy.sqrt`: Computes the square root of each element in the array.
  - Ceiling `numpy.ceil`: Rounds each element of the array to the nearest integer greater than or equal to that element.
  - Floor `numpy.floor`: Rounds each element of the array to the nearest integer less than or equal to that element.
  - Rounding `numpy.round`: Rounds each element of the array to the nearest integer.
```

      
  ### 5. Statistical Functions
  

>       - Mean (numpy.mean): Computes the arithmetic mean along the specified axis.
>       - Median (numpy.median): Computes the median along the specified axis.
>       - Standard Deviation (numpy.std): Computes the standard deviation along the specified axis.
>       - Variance (numpy.var): Computes the variance along the specified axis.

Reviewer: ```suggestion
  - Mean `numpy.mean`: Computes the arithmetic mean along the specified axis.
  - Median `numpy.median`: Computes the median along the specified axis.
  - Standard Deviation `numpy.std`: Computes the standard deviation along the specified axis.
  - Variance `numpy.var`: Computes the variance along the specified axis.
```

  
  ### 6. Linear Algebra
  

>       - Matrix Multiplication (numpy.matmul): Performs matrix multiplication between two arrays.
>       - Dot Product (numpy.dot): Computes the dot product of two arrays.
>       - Transpose (numpy.transpose): Computes the transpose of an array.

Reviewer: ```suggestion
  - Matrix Multiplication `numpy.matmul`: Performs matrix multiplication between two arrays.
  - Dot Product `numpy.dot`: Computes the dot product of two arrays.
  - Transpose `numpy.transpose`: Computes the transpose of an array.
```

> 
> 
> 
> 

Reviewer: ```suggestion
```
